### PR TITLE
Update Material to new warehouse design

### PR DESCRIPTION
**Pull request by aeslaughter**
(refs #5680)

### CODE REVIEW

**Comment by aeslaughter** [comment]
I have BISON and YAK patches waiting.

File: framework/src/base/BoundaryRestrictable.C [hunk 1]
```diff
@@ -256,10 +256,10 @@ BoundaryRestrictable::hasBoundaryMaterialPropertyHelper(const std::string & prop
     std::set<std::string> declared_props;
 
     // If boundary materials exist, populated the set of properties that were declared
-    if (material_warehouse.hasBoundaryMaterials(*id_it))
+    if (warehouse.hasActiveBoundaryObjects(*id_it))
     {
-      std::vector<Material *> mats = material_warehouse.getBoundaryMaterials(*id_it);
-      for (std::vector<Material *>::iterator mat_it = mats.begin(); mat_it != mats.end(); ++mat_it)
+      std::vector<MooseSharedPointer<Material> > mats = warehouse.getActiveBoundaryObjects(*id_it);
```
Comment (permcody) [inline framework/src/base/BoundaryRestrictable.C]: Seems like you are pretty good at switching these to `const`. You either missed one, or I'm failing to see why this one can't be `const`

**Comment by permcody** [comment]
Looks good, I only had two comments which should be addressed.
:+1: on the decent line reduction in this PR

**Comment by permcody** [comment]
@aeslaughter - I'm removing the "ready to merge" tag for now. With the current threading issues that you are fixing #6142 and the libMesh update, you'll want to rebase and repush this before we merge.

**Comment by permcody** [comment]
See that fail there? @aeslaughter said "merge it, it's fine"... Ok, I'll merge then ;)

**Pull request merged**